### PR TITLE
[SolYul] Storage mappings

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `51568deefaaf79c7fb940a07e4d3e5b833f5b86d`:
```
Error: Trailing whitespace found:
 test/libsolidity/semanticTests/viaYul/string_literals.sol:18:// short_dyn(): -> 
 test/libsolidity/semanticTests/viaYul/string_literals.sol:19:// long_dyn(): -> 

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `69d1a7674d7d1a6fa1561ec46b5128e66b421d48`:
```
Error: Trailing whitespace found:
 test/libsolidity/semanticTests/viaYul/string_literals.sol:18:// short_dyn(): -> 
 test/libsolidity/semanticTests/viaYul/string_literals.sol:19:// long_dyn(): -> 

```
Please check that your changes are working as intended.

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -539,6 +539,51 @@ string YulUtilFunctions::nextArrayElementFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::mappingIndexAccessFunction(MappingType const& _mappingType, Type const& _keyType)
+{
+	solAssert(_keyType.sizeOnStack() <= 1, "");
+
+	string functionName = "mapping_index_access_" + _mappingType.identifier() + "_of_" + _keyType.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		if (_mappingType.keyType()->isDynamicallySized())
```
Comment (ekpyron) [inline libsolidity/codegen/YulUtilFunctions.cpp:548]: This is very similar to ``packedHashFunction`` in #6663. The only difference is adding the base slot, right? Is this here the same as #6663's ``packedHashFunction`` with an appended ``uint256`` type and the slot as additional argument?

Comment (chriseth) [reply libsolidity/codegen/YulUtilFunctions.cpp:548]: Actually right, I could just use `tupleEncoderPacked` for the slot, too...

Comment (chriseth) [reply libsolidity/codegen/YulUtilFunctions.cpp:548]: Ah now I see what you mean - even packedHashFunction already supports everything!

Comment (ekpyron) [reply libsolidity/codegen/YulUtilFunctions.cpp:548]: Yes, that's what I meant!

**Review by ekpyron — APPROVED** [review]

**Pull request merged**